studio/src/hooks/useStickyLoading.ts

```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from "react";
+
+/**
+ * Makes the loading state sticky (for a minimum duration)
+ */
+export function useStickyLoading(loading: boolean, duration = 300) {
+  const [sticky, setSticky] = useState(false);
+
+  useEffect(() => {
+    if (loading) {
+      setSticky(true);
+      return;
+    }
+    const timeout = setTimeout(() => {
+      setSticky(false);
+    }, duration);
+    return () => clearTimeout(timeout);
+  }, [loading, duration]);
+
+  return sticky;
```

Comment:
should this return `sticky || loading`? 

if i understand correctly, it is a minimum duration for loading, but if `loading` is still true after `duration`, then it should still return `true`